consensus/src/quorum_store/batch_proof_queue.rs

```diff
@@ -759,9 +765,14 @@ impl BatchProofQueue {
                         }
                     }
                 }
-                item.proof = None;
-                item.proof_insertion_time = None;
-                item.txn_summaries = None;
+                item.mark_committed();
+            } else {
+                self.items.insert(batch_key, QueueItem {
```

Comment:
When does this item get removed?

Reply:
Good point, I don't see the original code handling it as well. Fixed and increased test coverage.